object_manager.go

```diff
@@ -454,6 +491,102 @@ func (objMgr *ObjectManager) DeleteHostRecord(ref string) (string, error) {
 	return objMgr.connector.DeleteObject(ref)
 }
 
+func (objMgr *ObjectManager) CreateARecord(netview string, dnsview string, recordname string, cidr string, ipAddr string, vmID string, vmName string) (*RecordA, error) {
+
+	ea := objMgr.getBasicEA(true)
+	ea["VM ID"] = "N/A"
+	ea["VM Name"] = "N/A"
+	if vmID != "" && vmName != "" {
+		ea["VM ID"] = vmID
+		ea["VM Name"] = vmName
+	}
+
+	recordA := NewRecordA(RecordA{
+		View:   dnsview,
+		Name:   recordname,
+		Ea:          ea})
+
+	if ipAddr == "" {
+		recordA.Ipv4Addr = fmt.Sprintf("func:nextavailableip:%s,%s", cidr, netview)
+	} else {
+		recordA.Ipv4Addr = ipAddr
+	}
+	ref, err := objMgr.connector.CreateObject(recordA)
+	recordA.Ref = ref
+	err = objMgr.connector.GetObject(recordA, ref, &recordA)
```

Comment:
Wouldn't `GetObject()` be ONLY necessary if `ipAddr` passed in is `""`? 

IOW, in the case if `ipAddr` is specified, wouldn't the resulting `recordA` would be no different from the `recordA` before calling `GetObject()`?

Reply:
yes, makes sense .Let me test it out without GetObject()

Reply:
actually it is not necessary even when ipaddress is passed empty